correct_periscope_drift.py

```diff
@@ -0,0 +1,431 @@
+# into a separate namespace, so that you can find out
+# where a routine or symbol is defined
+#
+
+
+from ciao_contrib.logger_wrapper import initialize_logger, make_verbose_level, set_verbosity, handle_ciao_errors
+from ciao_contrib.param_wrapper import open_param_file
+from ciao_contrib.runtool import dmstat
+import pycrates
+from pychips import (add_curve, print_window, set_plot_xlabel, set_plot_ylabel, clear_plot,
+                     add_window, set_plot_title)
+from sherpa import ui
+
+
+# Set up the logging/verbose code
+initialize_logger(TOOLNAME)
+
+# Use v<n> to display messages at the given verbose level.
+# You can pick other names than v0,v1, ... v5 if desired.
+#
+v1 = make_verbose_level(TOOLNAME, 1)
+v2 = make_verbose_level(TOOLNAME, 2)
+v5 = make_verbose_level(TOOLNAME, 5)
+
+
+def process_command_line(argv):
+
+    import paramio as pio
+    """Handle the parameter input for this script."""
+
+    if argv is None or argv == []:
+        raise ValueError("argv argument is None or empty")
+
+    # open_param_file, from ciao_contrib.param_wrapper,
+    # opens the parameter file and sorts out the command
+    # line. It returns a dictionary with a few fields;
+    # fp is the handle to use in paramio calls, as shown
+    # below; and parnams is the name to use if you need
+    # to re-open the parameter file to set things
+    # (which is unlikely, speak to Doug if you think you
+    # might need this).
+    #
+    pinfo = open_param_file(argv, toolname=TOOLNAME)
+    fp = pinfo["fp"]
+
+    # Use the parameter library to get the arguments
+    # and perform any desired validity checks (e.g.
+    # the outfile check/clobber check shown below).
+    #
+    # Parameters should be queried in the same order
+    # as the parameter file
+    #
+    mypars = {'progname': pinfo['progname'],
+              'parname': pinfo['parname']}
+
+    for stringpar in ['infile', 'evtfile',
+                      'outfile', 'corr_plot_prefix']:
+        mypars[stringpar] = pio.pgetstr(fp, stringpar)
+        if mypars[stringpar].strip() == "":
+            raise ValueError("{} parameter is empty".format(stringpar))
+
+    for floatpar in ['x', 'y', 'radius', 'src_min_counts']:
+        mypars[floatpar] = pio.pgetd(fp, floatpar)
+
+    mypars['corr_poly_degree'] = pio.pgeti(fp, "corr_poly_degree")
+
+    clobber = pio.pgetb(fp, "clobber")
+    verbose = pio.pgeti(fp, "verbose")
+
+    # We close the parameter file here; if you need to write
+    # values to the file you can either leave it open or -
+    # possibly better - is to close it and re-open it later.
+    #
+    pio.paramclose(fp)
+
+    # Set tool and module verbosity
+    set_verbosity(verbose)
+
+    # check outfile, exiting if it exists and clobber is
+    # not set. Note: the error message should be updated to better
+    # match that used by CIAO tools but I have not done this.
+    #
+    #
+    if not clobber and os.path.exists(mypars['outfile']):
```

Comment:
Not required, but we do have a "standard" clobber routine

```python
    from ciao_contrib._tools.fileio import outfile_clobber_checks
    outfile_clobber_checks(pars["clobber"], pars["outfile"] )
```
that you might want to use.